### PR TITLE
Adjust Boot 3 upgrade to use Boot 3 release version

### PR DESCRIPTION
**Pull request by BoykoAlex**

### CODE REVIEW

File: src/main/resources/META-INF/rewrite/spring-boot-30.yml [hunk 1]
```diff
@@ -113,8 +113,8 @@ recipeList:
 ########################################################################################################################
 # SpringBoot 3.0 Configuration Changes
 type: specs.openrewrite.org/v1beta/recipe
-name: org.openrewrite.java.spring.boot3.SpringBootProperties_3_0_0
-displayName: Migrate Spring Boot properties to 3.0.0
+name: org.openrewrite.java.spring.boot3.SpringBootProperties_3_0
```
Comment (tkvangorder) [inline src/main/resources/META-INF/rewrite/spring-boot-30.yml:116]: Hi, @BoykoAlex Thanks for the PR. 

We recently started [automating our change log (of recipes](https://docs.openrewrite.org/changelog/7-33-0-release)) and it made us painfully aware that we are too often making changes to the actual recipe names which will break downstream projects that rely on those names.

We can change the displayName, as this won't impact our users, but can you please revert the actual name of the recipe back to `org.openrewrite.java.spring.boot3.SpringBootProperties_3_0_0` ?

We have had some internal discussions about adding recipe aliases to make this type of change safely, but we have not settled on an implementation.

File: src/main/resources/META-INF/rewrite/spring-boot-30.yml [hunk 2]
```diff
@@ -33,7 +33,7 @@ recipeList:
   - org.openrewrite.java.spring.boot3.LegacyJmxExposure
   - org.openrewrite.java.spring.boot3.ActuatorEndpointSanitization
   - org.openrewrite.java.spring.boot3.Saml
-  - org.openrewrite.java.spring.boot3.SpringBootProperties_3_0_0
+  - org.openrewrite.java.spring.boot3.SpringBootProperties_3_0
```
Comment (tkvangorder) [inline src/main/resources/META-INF/rewrite/spring-boot-30.yml:36]: See comment below, we cannot make this change to the recipe name itself (without possibly breaking downstream users that are directly calling this recipe)

**Review by tkvangorder — CHANGES_REQUESTED** [review]

File: src/main/resources/META-INF/rewrite/spring-boot-30.yml (hunk 1 above)
Comment (BoykoAlex) [reply src/main/resources/META-INF/rewrite/spring-boot-30.yml]: Yes, makes sense. I reverted recipe id change :-)

**Review by tkvangorder — APPROVED** [review]
LGTM. Thank you!

File: src/main/resources/META-INF/rewrite/spring-boot-30.yml (hunk 2 above)
Comment (BoykoAlex) [reply src/main/resources/META-INF/rewrite/spring-boot-30.yml]: Sounds good - revert this change

**Pull request merged**